ltm.go

```diff
@@ -763,13 +806,255 @@ type monitorDTO struct {
+	DefaultsFrom          string
+	ExplicitFlowMigration string
+	HardwareSynCookie     string
+	IdleTimeout           string
+	ClientTimeout         int
+	IpTosToClient         string
+	IpTosToServer         string
+	KeepAliveInterval     string
+}
+
+type httpcompressDTO struct {
+	Name         string   `json:"name,omitempty"`
+	DefaultsFrom string   `json:"defaultsFrom,omitempty"`
+	UriExclude   []string `json:"uriExclude,omitempty"`
+	UriInclude   []string `json:"uriInclude,omitempty"`
+}
+
+type Httpcompresss struct {
+	Httpcompresss []Httpcompress `json:"items"`
+}
+
+type Httpcompress struct {
+	Name         string
+	DefaultsFrom string
+	UriExclude   []string
+	UriInclude   []string
+}
+
+type http2DTO struct {
+	Name                           string   `json:"name,omitempty"`
+	DefaultsFrom                   string   `json:"defaultsFrom,omitempty"`
+	ConcurrentStreamsPerConnection int      `json:"concurrentStreamsPerConnection,omitempty"`
+	ConnectionIdleTimeout          int      `json:"connectionIdleTimeout,omitempty"`
+	HeaderTableSize                int      `json:"headerTableSize,omitempty"`
+	ActivationModes                []string `json:"activationModes,omitempty"`
+}
+
+type Http2s struct {
+	Http2s []Http2 `json:"items"`
+}
+
+type Http2 struct {
+	Name                           string
+	DefaultsFrom                   string
+	ConcurrentStreamsPerConnection int
+	ConnectionIdleTimeout          int
+	HeaderTableSize                int
+	ActivationModes                []string
+}
+
+type Recordss struct {
+	Recordss []Records `json:"items"`
+}
+
+type Records struct {
+	Name string
+	Data string
+}
+type RecordsDTO struct {
+	Name string `json:"name,omitempty"`
+	Data string `json:"data,omitempty"`
+}
+
+type Datagroup struct {
+	Name    string
+	Type    string
+	Records []Records
+}
+
+type Datagroups struct {
+	Datagroups []Datagroup `json:"items"`
+}
+
+type DatagroupDTO struct {
+	Name    string `json:"name,omitempty"`
+	Type    string `json:"type,omitempty"`
+	Records struct {
+		Items []Records `json:"items,omitempty"`
+	} `json:"records,omitempty"`
+}
+
+func (p *Datagroup) MarshalJSON() ([]byte, error) {
+	var dto DatagroupDTO
+	marshal(&dto, p)
+	return json.Marshal(dto)
+}
+
+func (p *Datagroup) UnmarshalJSON(b []byte) error {
+	var dto DatagroupDTO
+	err := json.Unmarshal(b, &dto)
+	if err != nil {
+		return err
+	}
+	return marshal(p, &dto)
+}
+
+func (p *Records) MarshalJSON() ([]byte, error) {
+	var dto RecordsDTO
+	marshal(&dto, p)
+	return json.Marshal(dto)
+}
+
+func (p *Records) UnmarshalJSON(b []byte) error {
+	var dto RecordsDTO
+	err := json.Unmarshal(b, &dto)
+	if err != nil {
+		return err
+	}
+	return marshal(p, &dto)
+}
+
 func (p *Monitor) MarshalJSON() ([]byte, error) {
 	var dto monitorDTO
 	marshal(&dto, p)
 	if strings.Contains(dto.SendString, "\r\n") {
 		dto.SendString = strings.Replace(dto.SendString, "\r\n", "\\r\\n", -1)
 	}
-	return jsonMarshal(dto)
+	return json.Marshal(dto)
```

Comment:
Removing special escape processing https://github.com/scottdware/go-bigip/blob/master/bigip.go#L304 likely caused the test failure at https://travis-ci.org/scottdware/go-bigip/jobs/301672673

```
    --- FAIL: TestLtmSuite/TestCreateMonitorSpecialCharacters (0.00s)
	Error Trace:	ltm_test.go:817
	Error:      	Not equal: 
	            	expected: "{\"name\":\"test-web-monitor\",\"defaultsFrom\":\"http\",\"interval\":15,\"manualResume\":\"disabled\",\"recv\":\"Response &<>\",\"reverse\":\"disabled\",\"send\":\"GET /test&parms=1<2>3\\\\r\\\\n\",\"timeout\":5,\"transparent\":\"disabled\"}"
	            	actual: "{\"name\":\"test-web-monitor\",\"defaultsFrom\":\"http\",\"interval\":15,\"manualResume\":\"disabled\",\"recv\":\"Response \\u0026\\u003c\\u003e\",\"reverse\":\"disabled\",\"send\":\"GET /test\\u0026parms=1\\u003c2\\u003e3\\\\r\\\\n\",\"timeout\":5,\"transparent\":\"disabled\"}"
```
@scottdware